### DISCUSSION TITLE
To run two shells, *shell or *cmd

### DISCUSSION THREAD

**Discussion by jnz86** [MaJerle/lwshell#10] [Q&A]
I want to have my boot loader access a shell. Then also have a completely separate shell for working with an external module, where the host becomes a UART bridge. 

It seems a lot of the work was done to accept *lw object, but that it always gets the address from the static `shell` variable. It’s almost ready for two shells, but not quite.

I figure I could create two shell_t variables, or I could alternate in a separate list of commands for the second shell. Thoughts on the better approach?

Thanks!

ps> Can you elaborate on why out_fn() takes in *lw at all; and why it’s in reverse order of every other function in the code?

**Comment by MaJerle** [comment]
The `_ex` functions are about to be added, with option to provide shell instance. Similar like it is done in my https://github.com/MaJerle/lwprintf
